core/src/test/scala/org/typelevel/paiges/PaigesScalacheckTest.scala

```diff
@@ -212,50 +212,17 @@ class PaigesScalacheckTest extends FunSuite {
   }
 
   test("the left and right side of a union are the same after flattening") {
-    import Doc._
-    def okay(d: Doc): Boolean = d match {
-      case Empty | Text(_) | Line(_) => true
-      case Concat(a, b) => okay(a) && okay(b)
-      case Nest(j, d) => okay(d)
-      case Align(d) => okay(d)
-      case LazyDoc(d) => okay(d.evaluated)
-      case Union(a, b) =>
-        (a.flatten === b.flatten) && okay(a) && okay(b)
-    }
-
-    forAll { (a: Doc) => assert(okay(a)) }
+    forAll { u: Doc.Union => assert(u.a.flatten === u.b.flatten) }
   }
 
-  test("the left side of a union has a next line as long or longer than the right") {
-    import Doc._
-
-    def nextLineLength(d: Doc): (Boolean, Int) = d match {
-      case Line(_) => (true, 0)
-      case Empty => (false, 0)
-      case Text(s) => (false, s.length)
-      case Nest(j, d) => nextLineLength(d) // nesteding only matters AFTER the next line
-      case Align(d) => nextLineLength(d) // aligning only matters AFTER the next line
-      case Concat(a, b) =>
-        val r1@(done, l) = nextLineLength(a)
-        if (!done) {
-          val (d2, l2) = nextLineLength(b)
-          (d2, l2 + l)
-        } else r1
-      case LazyDoc(d) => nextLineLength(d.evaluated)
-      case Union(a, _) => nextLineLength(a) // assume the property is true
-    }
-
-    def okay(d: Doc): Boolean = d match {
-      case Empty | Text(_) | Line(_) => true
-      case Nest(j, d) => okay(d)
-      case Align(d) => okay(d)
-      case Concat(a, b) => okay(a) && okay(b)
-      case LazyDoc(d) => okay(d.evaluated)
-      case Union(a, b) =>
-        nextLineLength(a)._2 >= nextLineLength(b)._2
-    }
-
-    forAll { (a: Doc) => assert(okay(a)) }
+  test("the left side of a union has a first line as long or longer than the right") {
+    forAll(Gen.choose(1, 200), genUnion) { (n, u) =>
+      if (u.nonEmpty) {
+        def firstLine(d: Doc) = d.renderStream(n).head.takeWhile(_ != '\n')
```

Comment:
I don't think `head` works because the strings are concatenated out of the Stream:
note the comment:
https://github.com/typelevel/paiges/blob/master/core/src/main/scala/org/typelevel/paiges/Doc.scala#L243

So, I think we need to be a bit more careful here: if there is no `\n` in the first string, take the next, concat, and try again.

Reply:
Ug, thanks.  I'll put the loop back.  I was wondering why the test needed a loop, thinking each item was a line followed by the indent of the next line.

Reply:
Although, in 

```
-    def nextLineLength(d: Doc): (Boolean, Int) = d match {
-      case Line(_) => (true, 0)
-      case Empty => (false, 0)
-      case Text(s) => (false, s.length)
-      case Nest(j, d) => nextLineLength(d) // nesteding only matters AFTER the next line
-      case Align(d) => nextLineLength(d) // aligning only matters AFTER the next line
-      case Concat(a, b) =>
-        val r1@(done, l) = nextLineLength(a)
-        if (!done) {
-          val (d2, l2) = nextLineLength(b)
-          (d2, l2 + l)
-        } else r1
-      case LazyDoc(d) => nextLineLength(d.evaluated)
-      case Union(a, _) => nextLineLength(a) // assume the property is true
-    }
```

I can't understand why it's correct to ignore the rhs of a `Union`, since you don't know the width it's being printed at.